src/pkg/services/m365/api/mail.go

```diff
@@ -285,106 +285,179 @@ func (c Mail) GetItem(
+		}
 	}
 
-	attached, err := c.LargeItem.
+	size += totalSize
+
+	mail.SetAttachments(attachments)
+
+	return mail, MailInfo(mail, size), nil
+}
+
+// getAttachments attempts to get all attachments, including their content, in a singe query.
+func (c Mail) getAttachments(
+	ctx context.Context,
+	userID, mailID string,
+	immutableIDs bool,
+) ([]models.Attachmentable, int64, error) {
+	var (
+		result    = []models.Attachmentable{}
+		totalSize int64
+		cfg       = &users.ItemMessagesItemAttachmentsRequestBuilderGetRequestConfiguration{
+			QueryParameters: &users.ItemMessagesItemAttachmentsRequestBuilderGetQueryParameters{
+				Expand: []string{"microsoft.graph.itemattachment/item"},
+			},
+			Headers: newPreferHeaders(preferPageSize(maxNonDeltaPageSize), preferImmutableIDs(immutableIDs)),
+		}
+	)
+
+	attachments, err := c.LargeItem.
 		Client().
 		Users().
 		ByUserId(userID).
 		Messages().
-		ByMessageId(itemID).
+		ByMessageId(mailID).
 		Attachments().
-		Get(ctx, attachConfig)
-	if err == nil {
-		for _, a := range attached.GetValue() {
-			attachSize := ptr.Val(a.GetSize())
-			size += int64(attachSize)
-		}
-
-		mail.SetAttachments(attached.GetValue())
+		Get(ctx, cfg)
+	if err != nil {
+		return nil, 0, graph.Stack(ctx, err)
+	}
 
-		return mail, MailInfo(mail, size), nil
+	for _, a := range attachments.GetValue() {
+		totalSize += int64(ptr.Val(a.GetSize()))
+		result = append(result, a)
 	}
 
-	// A failure can be caused by having a lot of attachments as
-	// we are trying to fetch the data within the attachments as
-	// well in the request. We instead fetch all the attachment
-	// ids and fetch each item individually.
-	// NOTE: Maybe filter for specific error:
-	// graph.IsErrTimeout(err) || graph.IsServiceUnavailable(err)
-	// TODO: Once MS Graph fixes pagination for this, we can
-	// probably paginate and fetch items.
-	// https://learn.microsoft.com/en-us/answers/questions/1227026/pagination-not-working-when-fetching-message-attac
-	logger.CtxErr(ctx, err).Info("fetching all attachments by id")
+	return result, totalSize, nil
+}
 
-	// Getting size just to log in case of error
-	attachConfig.QueryParameters.Select = idAnd("size")
+// getAttachmentsIterated runs a two step fetch: one bulk query to get all attachment IDs,
+// and then another lookup to fetch the content of each attachment.
+// TODO: Once MS Graph fixes pagination for this, we can swap to a pager.
+// https://learn.microsoft.com/en-us/answers/questions/1227026/pagination-not-working-when-fetching-message-attac
+func (c Mail) getAttachmentsIterated(
+	ctx context.Context,
+	userID, mailID string,
+	immutableIDs bool,
+	errs *fault.Bus,
+) ([]models.Attachmentable, int64, error) {
+	var (
+		result    = []models.Attachmentable{}
+		totalSize int64
+		cfg       = &users.ItemMessagesItemAttachmentsRequestBuilderGetRequestConfiguration{
+			QueryParameters: &users.ItemMessagesItemAttachmentsRequestBuilderGetQueryParameters{
+				Select: idAnd(),
```

Comment:
We need size parameter too right?

Reply:
Nope, because this is just listing the IDs of each attachment.  We get the size when we look up the item itself on 392.